index.js

```diff
@@ -62,6 +63,8 @@ module.exports = function (opts) {
 				}
 
 				if (stat.isDirectory() || entry.fileName.charAt(entry.fileName.length - 1) === '/') {
+					new Mode(stat).isDirectory(true);
```

Comment:
yauzl didn't set correct mode to files. So, I want to be sure this directory will be a directory by stat.